File: problemas/matematica/1429/1429.c
#include <stdio.h>

int converte(int numero)
{
    int potencia = 1, p = 2, resposta = 0;

    while (numero)
    {
        resposta += potencia * (numero % 10);

        numero /= 10;
        potencia *= p++;
    }

    return resposta;
}

int main()
{
    int numero;

    while (scanf("%d", &numero))
    {
        if (!numero)
            break;

        printf("%d\n", converte(numero));
    }

    return 0;
}

File: problemas/matematica/1429/1429.php
<?php
function converteParaDecimal($numero) {
  $resposta = 0;
  $potencia = 0;
  $fatoriais = [1, 2, 6, 24, 120];

  for($i = strlen($numero) - 1; $i > -1; $i--) {
    $resposta += intval($numero[$i]) * $fatoriais[$potencia];
    $potencia += 1;
  }
  
  return $resposta;
}

while(!feof(STDIN)){
  $n = trim((string) fgets(STDIN));

  if ($n === "0") {
    break;
  }

  echo converteParaDecimal($n) . PHP_EOL;
}
?>